### PR TITLE
chore: remove new offer submission flag

### PR DESCRIPTION
**Pull request by jo-rs**
[NX-3015]

This PR removes flag checks for the "New Offer Submission" page and updates tests accordingly.
cc @artsy/negotiate-devs 

[NX-3015]: https://artsyproduct.atlassian.net/browse/NX-3015?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx [hunk 1]
```diff
@@ -24,6 +24,10 @@ jest.mock("v2/Utils/Events", () => ({
   postEvent: jest.fn(),
 }))
 
+jest.mock("v2/Utils/Hooks/useMatchMedia", () => ({
+  __internal__useMatchMedia: () => ({}),
+}))
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx:29]: Needed for the auto-scroll functionality not to error out

File: src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx [hunk 2]
```diff
@@ -183,7 +199,14 @@ describe("Offer InitialMutation", () => {
...
     })
 
+    it("doesn't let the user continue if they haven't clicked any option", async () => {
+      await page.clickSubmit()
+      expect(mutations.mockFetch).not.toHaveBeenCalled()
+      expect(page.offerInput.text()).toMatch("Offer amount missing or invalid.")
+    })
+
     it("doesn't let the user continue if they haven't typed anything in", async () => {
+      page.selectCustomAmount()
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx:209]: From now on the `OfferInput` is not visible until the user clicks on the custom option. So this first step is needed.

File: src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx [hunk 3]
```diff
@@ -286,10 +316,9 @@ describe("Offer InitialMutation", () => {
 
     it("tracks the offer input focus", () => {
       expect(mockPostEvent).not.toHaveBeenCalled()
-
+      page.selectCustomAmount()
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx:319]: From now on the OfferInput is not visible until the user clicks on the custom option. So this first step is needed.

File: src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx [hunk 4]
```diff
@@ -298,31 +327,33 @@ describe("Offer InitialMutation", () => {
...
     it("tracks viwing the low offer speedbump", async () => {
+      const trackData = {
+        action_type: "Viewed offer too low",
+        flow: "Make offer",
+        order_id: "1234",
+      }
       await page.setOfferAmount(1000)
 
-      expect(mockPostEvent).not.toHaveBeenCalled()
+      expect(mockPostEvent).not.toHaveBeenLastCalledWith(trackData)
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx:337]: We have a few tracking events firing now (see tests for `PriceOptions` component), so we have to be more specific when checking.

File: src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx [hunk 5]
```diff
@@ -286,10 +316,9 @@ describe("Offer InitialMutation", () => {
 
     it("tracks the offer input focus", () => {
       expect(mockPostEvent).not.toHaveBeenCalled()
-
+      page.selectCustomAmount()
       page.find("input").simulate("focus")
 
-      expect(mockPostEvent).toHaveBeenCalledTimes(1)
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx:292]: We would be now calling 3 tracking events by this point (see tests for `PriceOptions` component).

File: src/v2/Apps/Order/Routes/__tests__/Utils/OrderAppTestPage.tsx [hunk 6]
```diff
@@ -114,6 +133,7 @@ export class OrderAppTestPage extends RootTestPage {
   }
 
   async setOfferAmount(amount: number) {
+    if (this.find(PriceOptions).length) this.selectCustomAmount()
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/__tests__/Utils/OrderAppTestPage.tsx:136]: From now on the `OfferInput` is not visible until the user clicks on the custom option. So this first step is needed.

File: src/v2/Apps/__tests__/Fixtures/Order.ts [hunk 7]
```diff
@@ -66,7 +66,7 @@ const OrderArtworkNodeWithoutShipping = {
   is_offerable: true as boolean,
   listPrice: {
     __typename: "Money",
-    major: 10000,
+    major: 16000,
```
Comment (jo-rs) [inline src/v2/Apps/__tests__/Fixtures/Order.ts:69]: I think we didn't use this before, but now that we are, it was inconsistent with the total price marked on the file.

**Review by araujobarret — APPROVED** [review]

**Pull request merged**